scripts/ansible

```diff
@@ -1,106 +1,135 @@
+    FOUND="yes"
+    FAMILY="debian"
+fi
+if [ ! $FOUND = "yes" ]; then
+    if [ `grep -qi ubuntu /etc/lsb-release` ] ||  [ `grep -qi ubuntu /etc/os-release` ]; then
+        echo "deb http://ppa.launchpad.net/ansible/ansible/ubuntu xenial main" >> /etc/apt/sources.list
+        apt-get -y install dirmngr
+        apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 93C4A3FD7BB9C367
+        apt-get -y update
+#confirm PPA location
+#        apt-get install software-properties-common
+#        apt-add-repository ppa:ansible/ansible
+#        apt-get update
+        apt-get -y install ansible git python-pip python-setuptools python-wheel patch
+#        apt-get install ansible python-kerberos python-selinux python-winrm python-xmltodict sshpass bzip2 file findutils gzip tar unzip zip python-keyczar python-boto python-dnspython python-pyrax python-sphere
+        FOUND="yes"
+        FAMILY="debian"
     fi
+fi
+if [ ! $FOUND = "yes" ]; then
+    echo 'WARN: Could not detect distro or distro unsupported'
+    exit 1
+fi
 
-    # Install passlib for encrypt
-    yum -y groupinstall "Development tools"
-    yum -y install python-devel MySQL-python sshpass && pip install pyrax pysphere boto passlib dnspython
+# latest pip 2.2 is 2.2.3.0 on 2017-07-07
+# ansible-2.3.1.0-1.el7.noarch.rpm from 2017-06-01
 
-    # Install Ansible module dependencies
-    yum -y install bzip2 file findutils git gzip hg svn sudo tar which unzip xz zip libselinux-python
-    [ -n "$(yum search procps-ng)" ] && yum -y install procps-ng || yum -y install procps
-  elif [ -f /etc/debian_version ] || [ grep -qi ubuntu /etc/lsb-release ] || grep -qi ubuntu /etc/os-release; then
-    apt-get update
-    # Install via package
-    # apt-get update && \
-    # apt-get install --no-install-recommends -y software-properties-common && \
-    # apt-add-repository ppa:ansible/ansible && \
-    # apt-get update && \
-    # apt-get install -y ansible
+### start ansible pip install TODO add venv location /opt/iiab/anisble
+if [ $FAMILY = "olpc" ]; then
+    pip install ansible==2.4.0 --disable-pip-version-check
+    VER=`ansible --version|head -n 1|cut -f 2 -d " "`
+    echo "ansible version installed via pip $VER"
+fi
 
-    # Install required Python libs and pip
-    apt-get install -y  python-pip python-yaml python-jinja2 python-httplib2 python-paramiko python-pkg-resources
-    [ -n "$( apt-cache search python-keyczar )" ] && apt-get install -y  python-keyczar
-    if ! apt-get install -y git ; then
-      apt-get install -y git-core
-    fi
-    # If python-pip install failed and setuptools exists, try that
-    if [ -z "$(which pip)" -a -z "$(which easy_install)" ]; then
-      apt-get -y install python-setuptools
-      easy_install pip
-    elif [ -z "$(which pip)" -a -n "$(which easy_install)" ]; then
-      easy_install pip
+#  handle 2.2.1 2.2.0.0 <-> 2.4.0
+if [ $FAMILY = "debian" ]; then
+    if [ $GOOD_VER -lt 1 ]; then
```

Comment:
The error "line 110: [: -lt: unary operator expected" below implies that $GOOD_VER was never set (or at least has no value!) here on Raspbian anyway &mdash; when running http://download.iiab.io/6.4/rpi/load-ansible2.4.0-vpn.txt :

```
Setting up python-paramiko (2.0.0-1) ...
Setting up ansible (2.4.0.0-1ppa~xenial) ...
Processing triggers for libc-bin (2.24-11+deb9u1) ...
./scripts/ansible: line 110: [: -lt: unary operator expected
Current ansible version installed is 2.4.0.0
Running local playbooks!

PLAY [all] *********************************************************************
```

Reply:
Changed the logic a little please review